### PR TITLE
Fixes in registration list

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: indico/MaKaC/PDFinterface/conference.py [hunk 1]
```diff
@@ -1634,10 +1634,20 @@ def getBody(self, story=None, indexedFlowable={}, level=1 ):
...
             data = registration.data_by_field
             for item in self._display:
-                if item.input_type == 'accommodation' and item.id in data:
-                    lp.append(Paragraph(data[item.id].friendly_data.get('choice').encode('utf-8'), text_format))
-                    lp.append(Paragraph(format_date(data[item.id].friendly_data.get('arrival_date')), text_format))
-                    lp.append(Paragraph(format_date(data[item.id].friendly_data.get('departure_date')), text_format))
+                if item.input_type == 'accommodation':
+                    if item.id in data:
+                        lp.append(Paragraph(data[item.id].friendly_data.get('choice').encode('utf-8'), text_format))
+                        arrival = data[item.id].friendly_data.get('arrival_date')
```
Comment (ThiefMaster) [inline indico/MaKaC/PDFinterface/conference.py]: why not use `arrival_date` here, too, and then overwrite the variable in the next line?

File: indico/modules/events/registration/templates/display/_registration_summary_blocks.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {% macro render_registration_info(registration, from_management=false) %}
-    {% for section in registration.registration_form.sections -%}
+    {% for section in registration.registration_form.sections if from_management or not section.is_manager_only -%}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/_registration_summary_blocks.html]: We need this check in the email, too

File: indico/modules/events/registration/templates/display/participant_list.html [hunk 3]
```diff
@@ -4,6 +4,10 @@
     {% trans %}Participant List{% endtrans %}
 {% endblock %}
 
+{% block subtitle %}
+    {% trans num=registrations|length %}{{ num }} participants{% endtrans %}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/participant_list.html]: pluralize

Comment (nop33) [inline indico/modules/events/registration/templates/display/participant_list.html]: Hm, showing "1 participant" doesn't seem so nice.
How about "Total participants: 1"?

Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/participant_list.html]: "One participant"..

Comment (nop33) [inline indico/modules/events/registration/templates/display/participant_list.html]: I know how to do it, I was just wondering for the wording. I still don't line "One participant" as a subtitle...

Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/participant_list.html]: What about showing the subtitle only if there's more than one participant?

Comment (nop33) [inline indico/modules/events/registration/templates/display/participant_list.html]: :+1:

File: indico/modules/events/registration/templates/display/_registration_summary_blocks.html (hunk 2 above)
Comment (nop33) [inline indico/modules/events/registration/templates/display/_registration_summary_blocks.html]: Done.

File: indico/modules/events/registration/templates/display/participant_list.html [hunk 4]
```diff
@@ -4,6 +4,13 @@
     {% trans %}Participant List{% endtrans %}
 {% endblock %}
 
+{% block subtitle %}
+    {% set num_of_registations = registrations | length %}
+    {% if num_of_registations > 1 %}
+        {%- trans  %}{{ num_of_registations }} participants{% endtrans -%}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/participant_list.html]: extra space after trans

File: indico/modules/events/registration/templates/display/participant_list.html [hunk 5]
```diff
@@ -4,6 +4,13 @@
     {% trans %}Participant List{% endtrans %}
 {% endblock %}
 
+{% block subtitle %}
+    {% set num_of_registations = registrations | length %}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/participant_list.html]: why not just `num_registrations` (also missing a `r` btw) or even just `count`?

File: indico/modules/events/registration/templates/emails/base_registration_details.html [hunk 6]
```diff
@@ -1,7 +1,8 @@
 {% extends 'events/registration/emails/base_registration.html' %}
 
 {% block registration_body %}
-    {% for section, fields in registration.summary_data.iteritems() %}
+    {% for section, fields in registration.summary_data.iteritems()
+       if to_managers or not section.is_manager_only %}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/emails/base_registration_details.html]: Actually, on a second thought, I think we should always hide it here. It doesn't really belong in those emails at all

File: indico/modules/events/registration/templates/display/participant_list.html (hunk 5 above)
Comment (nop33) [inline indico/modules/events/registration/templates/display/participant_list.html]: :laughing: :laughing: :laughing:  regisTations!
I am fixing it..

File: indico/modules/events/registration/templates/emails/base_registration_details.html (hunk 6 above)
Comment (nop33) [inline indico/modules/events/registration/templates/emails/base_registration_details.html]: are u sure?

Comment (ThiefMaster) [inline indico/modules/events/registration/templates/emails/base_registration_details.html]: at least we never included it in emails before.. it's manager-only information anyway

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 7]
```diff
@@ -237,6 +242,34 @@ def _render_registration_list(regform, registrations, total_registrations=None):
...
+    # done sending it to the client.
+    delete_file.apply_async(args=[temp_file.name], countdown=3600)
+    temp_file.delete = False
+    return send_file('attachments-{}.zip'.format(regform.id), temp_file.name, 'application/zip', inline=False)
+
+
+def _prepare_folder_structure(attachment):
+    registration = attachment.registration
+    regform_title = secure_filename(attachment.registration.registration_form.title, 'registration_form')
+    registrant_name = secure_filename("{}_{}".format(registration.full_name, unicode(registration.friendly_id)),
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py]: `registration.get_full_name()`

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 8]
```diff
@@ -237,6 +242,34 @@ def _render_registration_list(regform, registrations, total_registrations=None):
...
+                                      registration.friendly_id)
+    file_name = secure_filename("{}_{}_{}".format(attachment.field_data.field.title, attachment.field_data.field_id,
+                                                  attachment.filename), attachment.filename)
+    path = os.path.join(*adjust_path_length(filter(None, [regform_title, registrant_name, file_name])))
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py]: no need for the filter

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 9]
```diff
@@ -567,3 +599,18 @@ def _process(self):
...
+
+class RHRegistrationsExportAttachments(RHRegistrationsExportBase):
+    """Export registration attachments in a zip file"""
+
+    def _process(self):
+        attachments = {}
+        file_fields = [item for item in self.regform.form_items if item.input_type == 'file']
+        for registration in self.registrations:
+            data = registration.data_by_field
+            attachments_for_registrant = [data.get(file_field.id) for file_field in file_fields
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py]: for *registration*

File: indico/modules/events/registration/templates/management/regform_reglist.html [hunk 10]
```diff
@@ -93,7 +93,13 @@
                             {%- trans %}Print Badges{% endtrans -%}
                         </a>
                     </li>
-                    <li><a href="#" class="icon-attachment js-requires-selected-row disabled">{% trans %}Attachments{% endtrans %}</a></li>
+                    <li>
+                        <a href="#"
+                           class="icon-attachment js-requires-selected-row disabled js-submit-reglist-form"
+                           data-href="{{ url_for('.registrations_attachments_export', regform) }}">
+                            {%- trans %}Downlaod attachments{% endtrans -%}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/management/regform_reglist.html]: Download

Comment (ThiefMaster) [inline indico/modules/events/registration/templates/management/regform_reglist.html]: Download

**Comment by ThiefMaster** [comment]
Merged dd61a62

**Pull request closed without merging**